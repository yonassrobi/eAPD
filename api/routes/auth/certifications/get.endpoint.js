import {
  getDB,
  setupDB,
  teardownDB,
  login,
  unauthenticatedTest,
  unauthorizedTest
} from '../../../endpoint-tests/utils.js';

describe('auth/certifications get endpoint', () => {
  describe('GET /auth/certifications', () => {
    const db = getDB();
    beforeAll(() => setupDB(db));
    afterAll(() => teardownDB(db));

    const url = '/auth/certifications';

    unauthenticatedTest('get', url);
    unauthorizedTest('get', url);

    describe('when authenticated as a user with permission', () => {
      let api;
      beforeAll(async () => {
        api = login('fed-admin');
      });

      it('valid request', async () => {
        const response = await api.get(url);

        expect(response.status).toEqual(200);
        expect(response.data).toMatchSnapshot();
      });
    });
  });
});
