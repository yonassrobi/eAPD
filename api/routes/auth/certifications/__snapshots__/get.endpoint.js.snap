// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`auth/certifications get endpoint GET /auth/certifications when authenticated as a user with permission valid request 1`] = `
Array [
  Object {
    "affiliationId": null,
    "email": "stateadminmatch@email.com",
    "ffy": 2023,
    "fileUrl": "http://localhost:8081/auth/certifications/files/test-123",
    "id": 5004,
    "name": "State Admin Test",
    "potentialMatches": "1",
    "state": "ak",
  },
]
`;
